### DISCUSSION TITLE
Color code 'success on retry'

### DISCUSSION THREAD

**Discussion by fehtemam** [apache/airflow#41248] [Ideas]
### Description

While there are tools to analyze logs and understand how many times tasks have failed before they succeed, it would be very informative to have separate color for tasks that succeed but not on their first try. So when a task succeeds on the first try it will have a dark green color as it has now but if it succeeds after failing initially, then it will have a different shade of green or blue or whatever color is the best. This gives a very quick yet very informative look of dagruns when we are looking at a history of 25, 50, 100 ... of them at once.

### Use case/motivation

What will be achieved here is making the color codes more informative than it already is by including extra information about dagruns history.

### Related issues

No

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by bbovenzi** [comment]
My only hesitation is that we already have a lot of colors that are already hard to distinguish at times. I would like to try to use iconography to also portray information, maybe this would be a good candidate.

**Comment by fehtemam** [comment]
I agree with the color palette already being saturated with too many hues. I think the idea of using separate icons is very interesting. Like showing a triangle instead of a square if the task is completed after retrying while failing on the first try. Although, that might add additional load to the UI to start the icon as one shape and update it using another shape after the task is finished based on the final status. Maybe we can have a square with its diagonal drawn inside from the start and if the task finishes on the first try then both halves of the square are shown as dark green but if it finishes with at least one retry then only the upper half of the square is filled with the corresponding success color. I'm thinking loud here. Also unfortunately I won't be able to contribute a PR but I thought sharing the idea for this feature might be better than not sharing it. Perhaps someone on the frontend with interest in this might be able to work on it in the future.

**Comment by bbovenzi** [comment]
In the updated Tree View we already use a "play" icon to indicate manually triggered runs instead of a border. I may play around with a retry icon inside a task instance itself.

<img width="329" alt="Screen Shot 2022-02-09 at 5 18 15 PM" src="https://user-images.githubusercontent.com/4600967/153299719-f245023c-6c75-422a-9c9d-445cf1769dff.png">

**Comment by potiuk** [comment]
> In the updated Tree View we already use a "play" icon to indicate manually triggered runs instead of a border. I may play around with a retry icon inside a task instance itself.

Another option is to use the shape of the task instance: 

![retry](https://user-images.githubusercontent.com/595491/153390990-ff152a37-7e13-46e2-940e-3582a81dbec6.png)

**Comment by eladkal** [comment]
>  it would be very informative to have separate color for tasks that succeed but not on their first try.

I feel that this is kind of information that should be presented in some kind of report/analytic system rather than on operational views.
